plugins/optimization-detective/detect.js

```diff
@@ -98,20 +98,31 @@ function error( ...message ) {
 /**
  * Checks whether the URL Metric(s) for the provided viewport width is needed.
  *
+ * The comparison logic here corresponds with the PHP logic in `OD_URL_Metric_Group::is_viewport_width_in_range()`.
+ *
  * @param {number}                 viewportWidth          - Current viewport width.
  * @param {URLMetricGroupStatus[]} urlMetricGroupStatuses - Viewport group statuses.
  * @return {boolean} Whether URL Metrics are needed.
  */
 function isViewportNeeded( viewportWidth, urlMetricGroupStatuses ) {
-	let lastWasLacking = false;
-	for ( const { minimumViewportWidth, complete } of urlMetricGroupStatuses ) {
-		if ( viewportWidth >= minimumViewportWidth ) {
-			lastWasLacking = ! complete;
-		} else {
-			break;
+	if ( viewportWidth === 0 ) {
+		return false;
+	}
+
+	for ( const {
+		minimumViewportWidth,
+		maximumViewportWidth,
+		complete,
+	} of urlMetricGroupStatuses ) {
+		if (
+			viewportWidth > minimumViewportWidth &&
```

Comment:
Nit-pick: Aren't the `urlMetricGroupStatuses` sorted by width? If we can rely on that, we could probably get rid of this condition, because it would be implied by the fact that the previous group didn't `return complete;` yet.

Reply:
They are sorted by minimum viewport width, yes. However, I thought it would be preferable to have the logic be identical as is found in `OD_URL_Metric_Group::is_viewport_width_in_range()`. It also seems easier to reason about than having to remember to factor in the loop and the fact that the groups are sorted.